api/src/main/java/com/datastrato/gravitino/tag/Tag.java

```diff
@@ -0,0 +1,110 @@
+ * fileset, topic, or column. It can be used to categorize, classify, or annotate these entities.
+ */
+@Evolving
+public interface Tag extends Auditable {
+
+  /**
+   * A reserved property to specify the color of the tag. The color is a string of hex code that
+   * represents the color of the tag. The color is used to visually distinguish the tag from other
+   * tags.
+   */
+  String PROPERTY_COLOR = "color";
+
+  /** @return The name of the tag. */
+  String name();
+
+  /** @return The comment of the tag. */
+  String comment();
+
+  /** @return The properties of the tag. */
+  Map<String, String> properties();
+
+  /**
+   * Check if the tag is inherited from a parent entity or not. If the tag is inherited, it will
+   * return true, if it is owned by the entity itself, it will return false.
+   *
+   * <p>Note. The return value is optional, only when the tag is associated with an entity, and
+   * called from the entity, the return value will be present. Otherwise, it will be empty.
+   *
+   * @return True if the tag is inherited, false if it is owned by the entity itself. Empty if the
+   *     tag is not associated with any entity.
+   */
+  Optional<Boolean> inherited();
+
+  /** @return Extended information of the tag. */
+  default Extended extended() {
+    throw new UnsupportedOperationException("The extended method is not supported.");
+  }
+
+  /** The extended information of the tag. */
+  interface Extended {
+
+    /** The supported entities that a tag can be attached to. */
+    enum SupportedEntityType {
+      /** Catalog type entity. */
+      CATALOG,
+      /** Schema type entity. */
+      SCHEMA,
+      /** Table type entity. */
+      TABLE,
+      /** Fileset type entity. */
+      FILESET,
+      /** Topic type entity. */
+      TOPIC,
+      /** Column type entity. */
+      COLUMN
+    }
+
+    /** A wrapper class to represent an entity that is using the tag. */
+    final class UsedEntity {
+      private final SupportedEntityType type;
+      private final String identifier;
+
+      private UsedEntity(SupportedEntityType type, String identifier) {
+        this.type = type;
+        this.identifier = identifier;
+      }
+
+      /** @return The type of the entity. */
+      public SupportedEntityType type() {
+        return type;
+      }
+
+      /** @return The String identifier of the entity. */
+      public String identifier() {
+        return identifier;
+      }
+
+      /**
+       * Create a new instance of the used entity.
+       *
+       * @param type The type of the entity.
+       * @param identifier The String identifier of the entity.
+       * @return The used entity.
+       */
+      public static UsedEntity of(SupportedEntityType type, String identifier) {
+        return new UsedEntity(type, identifier);
+      }
+    }
+
+    /** @return The number of entities that are using this tag. */
+    int inUse();
```

Comment:
should this method have a default implementation?
```java
return usedEntities().length
```

Reply:
I see.